lib/trino-filesystem-cache-alluxio/src/main/java/io/trino/filesystem/alluxio/TracingCacheManager.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.filesystem.alluxio;
+
+import alluxio.client.file.CacheContext;
+import alluxio.client.file.cache.CacheManager;
+import alluxio.client.file.cache.CacheUsage;
+import alluxio.client.file.cache.PageId;
+import alluxio.exception.PageNotFoundException;
+import alluxio.file.ReadTargetBuffer;
+import alluxio.network.protocol.databuffer.DataFileChannel;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+
+import java.nio.ByteBuffer;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import static io.trino.filesystem.alluxio.AlluxioTracing.withTracing;
+import static io.trino.filesystem.cache.CacheSystemAttributes.CACHE_FILE_READ_POSITION;
+import static io.trino.filesystem.cache.CacheSystemAttributes.CACHE_FILE_READ_SIZE;
+import static io.trino.filesystem.cache.CacheSystemAttributes.CACHE_FILE_WRITE_POSITION;
+import static io.trino.filesystem.cache.CacheSystemAttributes.CACHE_FILE_WRITE_SIZE;
+import static io.trino.filesystem.cache.CacheSystemAttributes.CACHE_KEY;
+import static java.util.Objects.requireNonNull;
+
+public class TracingCacheManager
+        implements CacheManager
+{
+    private final Tracer tracer;
+    private final String cacheKey;
+    private final CacheManager delegate;
+
+    public TracingCacheManager(Tracer tracer, String cacheKey, CacheManager delegate)
+    {
+        this.tracer = requireNonNull(tracer, "tracer is null");
+        this.cacheKey = requireNonNull(cacheKey, "cacheKey is null");
+        this.delegate = requireNonNull(delegate, "delegate is null");
+    }
+
+    @Override
+    public boolean put(PageId pageId, ByteBuffer byteBuffer, CacheContext cacheContext)
+    {
+        Span span = tracer.spanBuilder("AlluxioCacheManager.put")
+                .setAttribute(CACHE_KEY, cacheKey)
```

Comment:
Do we need `cacheKey` on these spans? It should be present on outer spans. It's also a bit strange that `cacheKey` is a field of a CacheManager, which in principle could put pages for multiple different `cacheKeys`.

Reply:
That's why I wrap `CacheManager` with `TracingCacheManager` in `AlluxioFileSystemCache` so I can track that information even though `CacheManager` is a singleton shared between multiple cached inputs/streams.

Reply:
That's part that makes sense for tests. Without that I wouldn't be able to correlate these operations with the higher level calls